routers/web/user/home.go

```diff
@@ -420,6 +422,36 @@ func buildIssueOverview(ctx *context.Context, unitType unit.Type) {
 		User:       ctx.User,
 	}
 
+	// Search all repositories which
+	//
+	// As user:
+	// - Owns the repository.
+	// - Have collaborator permissions in repository.
+	//
+	// As org:
+	// - Owns the repository.
+	//
+	// As team:
+	// - Team org's owns the repository.
+	// - Team has read permission to repository.
+	repoOpts := &models.SearchRepoOptions{
+		Actor:      ctx.User,
+		OwnerID:    ctx.User.ID,
+		Private:    true,
+		AllPublic:  false,
+		AllLimited: false,
+	}
+
+	if ctxUser.IsOrganization() && ctx.Org.Team != nil {
+		repoOpts.TeamID = ctx.Org.Team.ID
+	}
+
+	userRepoIDs, _, err := models.SearchRepositoryIDs(repoOpts)
```

Comment:
We should not really read all the IDs into memory, we could build a subquery condition via https://gitea.com/xorm/builder to make this better performance. I have sent a PR removed that. I don't think we need to take it back.

Reply:
Which PR?

Reply:
Can't find that one. But we really should avoid to load ids into the memory. You never know how many repositories one has.

Reply:
I think I fail to understand the importance why this should be avoided. I can think of far bigger things that are loaded into memory than just a slice of int. Unless somehow we need to handle use-cases whereby a user has a good 100K of repo's(which has it's own problems) I don't think it makes sense to optimize this away, there are other things in this particular codepath that will be quite larger.

If we really should avoid this, then this PR will become quite larger than it is already as it loading this kind of information into memory and then passing it as a value is quite normal, creating cross-functions-conditions seems to be new for Gitea.

Reply:
In fact, you can find `SearchRepositoryCondition` which will return a condition, you can use that condition directly.

Reply:
We can have a `RepoCond` field in `UserIssueStatsOptions` to store the condition.

Reply:
The `UserIssueStatsOptions` won't be a problem as that is one and all database queries.

The problem is rather with `issueIDsFromSearch` it gives `indexer.Search` the repo id's as a slice and looking at all the indexers, it isn't possible to get that working with database conditions.

Reply:
Yes, that's another problem we currently cannot handler.